### DISCUSSION TITLE
How to use `protoc-gen-ent` seamlessly

### DISCUSSION THREAD

**Discussion by kkweon** [ent/ent#1642] [Q&A] [answered]
According to this guide(https://entgo.io/blog/2021/05/04/announcing-schema-imports),
`protoc-gen-ent` creates a new `User` struct different from `User` struct generated by `protoc-gen-go`.

Now, there will be 2 `User` structs.

- One in `ent/proto/entpb` package
- One in `ent/schema` package

Is this how it's supposed to work?

## Expectation

My expectation is that it should override `User` generated by `protoc-gen-go`, so there is only one `User` struct in the file.
If I am going to keep two structs, I want the `ent` one to have a different name `UserDao` or something to make it explicit it's only used for the database ORM.

**Comment by rotemtam** [comment, marked as answer]
Hi
Thanks for writing. I
totally understand what you are saying. It must seem a bit duplicate to have two structs by the same name. However, as Rob Pike famously said, "a little copying is better than a little dependency". In other words , duplicating some parts in our software isn't as bad as coupling things together.

Here we have two structs would be wrong to lump together in my eyes. They are easily qualified by package name (entpb.User and ent.User) which is quite standard in Go. 

Adding a suffix  to either one would be redundant as they are already qualified and represented in a qualified way everywhere, don't you think?

**Reply by kkweon** [reply]
Thank you for explaining. Yes, I see the value separating it out.

But, sometimes, I do want to represent database schema in proto.

Anyway, I see your point. Thank you!
